wallet-new/cardano-sl-wallet-new.cabal

```diff
@@ -15,7 +15,51 @@ cabal-version:       >=1.10
 
 library
   hs-source-dirs:      src
-  exposed-modules:     Cardano.Wallet.TypeLits
+  exposed-modules:     Cardano.Wallet.API
```

Comment:
I moved all of the modules in `server/` to `src/` and made them `exposed-modules` in the library.

Reply:
I also sorted the list.

Reply:
Interesting choice -- the reason why I went with the original split was because (at least in my head), the way you implement your handlers is a sort of "implementation detail" which better lives in the `executable` realm, whereas the API itself (as in its definition and domain-specific types) is better suited in a `library`.

The major snag was that I had to specify in my tests both `server src` as `hs-source-dirs`, which was a bit annoying, admittedly.